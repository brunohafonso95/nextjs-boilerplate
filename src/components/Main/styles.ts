import styled from 'styled-components';

export const Wrapper = styled.div`
  align-items: center;
  background-color: aliceblue;
  display: flex;
  height: 100%;
  justify-content: center;
  width: 100%;
`;
