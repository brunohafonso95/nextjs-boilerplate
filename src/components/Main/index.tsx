import { MainProps } from './interfaces';
import * as S from './styles';

const Main = ({ title = 'NextJS Boilerplate' }: MainProps) => {
  return (
    <S.Wrapper>
      <h1>{title}</h1>
    </S.Wrapper>
  );
};
export default Main;
