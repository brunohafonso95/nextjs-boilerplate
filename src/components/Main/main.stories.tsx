import { ArgTypes, Meta, Story } from '@storybook/react';

import Main from '.';
import { MainProps } from './interfaces';

export default {
  title: 'Main',
  component: Main,
  parameters: {
    layout: 'fullscreen',
  },
  argTypes: {
    title: {
      name: 'title',
      type: 'string',
      description: 'the title text of the component',
    },
  } as ArgTypes<MainProps>,
} as Meta;

export const Basic: Story<MainProps> = (args) => <Main {...args} />;
Basic.args = {
  title: 'NextJS Boilerplate',
} as MainProps;
