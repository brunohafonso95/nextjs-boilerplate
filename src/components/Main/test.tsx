import { screen, render } from '@testing-library/react';

import Main from '.';

describe('Unit tests for Main Component', () => {
  it('should render a header element', () => {
    const { container } = render(<Main />);
    expect(screen.getByRole('heading', { name: /NextJS Boilerplate/ })).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyle({ 'background-color': 'aliceblue' });
  });
});
