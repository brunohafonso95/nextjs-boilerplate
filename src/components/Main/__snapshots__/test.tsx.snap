// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for Main Component should render a header element 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: aliceblue;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100vh;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100vw;
}

<div
  class="c0"
>
  <h1>
    NextJS Boilerplate
  </h1>
</div>
`;
